Resources/Private/JavaScript/InspectorEditors/Reference/index.js

```diff
@@ -0,0 +1,65 @@
+import React, {Component, PropTypes} from 'react';
+import SelectBox from '@neos-project/react-ui-components/lib/SelectBox/';
+import {connect} from 'react-redux';
+import {$transform, $get} from 'plow-js';
+
+const {ApiEndpoints} = window['@Neos:HostPluginAPI'];
+const {searchNodes} = ApiEndpoints;
+
+@connect($transform({
+    activeWorkspace: $get('cr.workspaces.active'),
+    siteNodePath: $get('cr.nodes.siteNode')
+}))
+export default class ReferenceEditor extends Component {
+
+    static propTypes = {
+        commit: PropTypes.func.isRequired,
+        value: PropTypes.any.isRequired,
+
+        options: PropTypes.shape({
+            nodeTypes: PropTypes.arrayOf(PropTypes.string)
+        }),
+
+        activeWorkspace: PropTypes.string.isRequired,
+        siteNodePath: PropTypes.string.isRequired
+    };
+
+    render() {
+        const {commit} = this.props;
+        const valueSoFar = this.props.value;
+
+        const options = ({value, callback}) => {
+            const searchQuery = {
+                // TODO: dimensons!
+                workspaceName: this.props.activeWorkspace,
+                contextNode: this.props.siteNodePath
+            };
+
+            if (!value && valueSoFar) {
+                searchQuery.nodeIdentifiers = [valueSoFar];
+            } else if (!value || value.length < 2) {
+                return;
+            } else {
+                searchQuery.searchTerm = value;
+            }
+
+            searchNodes(searchQuery)
+            .then(result => result.text())
+            .then(result => {
+                const d = document.createElement('div');
+                d.innerHTML = result;
+                const nodes = d.querySelector('.nodes');
```

Comment:
I'm afraid this pattern must be applied in a lot of places, due to the design of the current backend APIs.

I'd prefer a cleaner JSON approach as well, but it would be okay to do that at a later point. 

In the meantime, :+1: for moving this part to the API.